### ISSUE TITLE
Registering an fsspec backend breaks gcsfs

### ISSUE THREAD

**Issue by AlpAribal** [dask/gcsfs#423]
**What happened**: When a package [registers its fsspec backend](https://filesystem-spec.readthedocs.io/en/latest/developer.html#implementing-a-backend), `fsspec.spec` becomes an `EntryPoint` and hence classes under the spec module cannot be directly used (e.g. using `fsspec.spec.AbstractBufferedFile` raises an error). As a result, I cannot import gcsfs anymore since `GCSFile` definition uses `fsspec.spec.AbstractBufferedFile`.

**What you expected to happen**:
To be able to use gcsfs as usual.

**Minimal Complete Verifiable Example**:

```python
import gcsfs
```
raises the following error:
```python
    class GCSFile(fsspec.spec.AbstractBufferedFile):
AttributeError: 'EntryPoint' object has no attribute 'AbstractBufferedFile'
```

**Anything else we need to know?**:
This can be fixed by having the import `from fsspec.spec import AbstractBufferedFile` and then using `AbstractBufferedFile` when needed. I would be up for creating a PR for this fix.

**Environment**:

- Dask version: N/A
- Python version: 3.8.11
- Operating System: Ubuntu 20.04.3
- Install method (conda, pip, source): `pip install gcsfs==2021.10 adlfs==2021.10`

**Comment by martindurant** [comment]
This is fixed in https://github.com/intake/filesystem_spec/pull/784

**Comment by AlpAribal** [comment]
> This is fixed in [intake/filesystem_spec#784](https://github.com/intake/filesystem_spec/pull/784)

Great, thank you for the lightning reply!
